contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -0,0 +1,151 @@
+        require(address(_subscription) != address(0), "Subscription cannot be the zero address");
+        subscription = _subscription;
+    }
+
+    /**
+     * @notice Checks if the sender is the authority of the ritual
+     * @param ritualId The ID of the ritual
+     */
+    modifier onlyCohortAuthority(uint32 ritualId) {
+        require(
+            coordinator.getAuthority(ritualId) == msg.sender,
+            "Only cohort authority is permitted"
+        );
+        _;
+    }
+
+    /**
+     * @notice Checks if the sender has allowance to set authorizations
+     * @dev This function overrides the canSetAuthorizations modifier in the GlobalAllowList contract
+     * @param ritualId The ID of the ritual
+     */
+    modifier canSetAuthorizations(uint32 ritualId) override {
+        require(getAllowance(ritualId, msg.sender) > 0, "Only administrator is permitted");
+        _;
+    }
+
+    /**
+     * @notice Returns the allowance of an administrator for a ritual
+     * @param ritualId The ID of the ritual
+     * @param admin The address of the administrator
+     * @return The allowance of the administrator
+     */
+    function getAllowance(uint32 ritualId, address admin) public view returns (uint256) {
+        return allowance[LookupKey.lookupKey(ritualId, admin)];
+    }
+
+    /**
+     * @dev This function is called before the setAuthorizations function
+     * @param ritualId The ID of the ritual
+     * @param addresses The addresses to be authorized
+     * @param value The authorization status
+     */
+    function _beforeSetAuthorization(
+        uint32 ritualId,
+        address[] calldata addresses,
+        // TODO: Currently unused, remove?
+        // solhint-disable-next-line no-unused-vars
+        bool value
+    ) internal view override {
+        for (uint256 i = 0; i < addresses.length; i++) {
+            require(
+                authActions[ritualId] <
+                    subscription.authorizationActionsCap(ritualId, addresses[i]),
+                "Authorization cap exceeded"
+            );
+        }
+    }
+
+    /**
+     * @notice Sets the administrator caps for a ritual
+     * @dev Only active rituals can set administrator caps
+     * @param ritualId The ID of the ritual
+     * @param addresses The addresses of the administrators
+     * @param value The cap value
+     */
+    function setAdministratorCaps(
+        uint32 ritualId,
+        address[] calldata addresses,
+        uint256 value
+    ) internal {
+        require(
+            coordinator.isRitualActive(ritualId),
+            "Only active rituals can set administrator caps"
+        );
+        for (uint256 i = 0; i < addresses.length; i++) {
+            allowance[LookupKey.lookupKey(ritualId, addresses[i])] = value;
+            emit AdministratorCapSet(ritualId, addresses[i], value);
+        }
+        authActions[ritualId] += addresses.length;
```

Comment:
Incrementing if `value` is `True`. Should we be decrementing if `value` is `False`?